README.md

```diff
@@ -20,24 +20,90 @@ Specification for binding services to runtime applications running in Kubernetes
+#### Minimum
+For a service to be bindable it **MUST** either:
+* provide a Secret that contains the binding data and reference this Secret using one of the patterns discussed [below](#pointer-to-binding-data). 
+* map its `status` properties to the corresponding binding data, using one of the patterns discussed [below](#pointer-to-binding-data).
 
-### 2.  Service Binding Schema
+To allow existing services to be bindable with zero changes, implementers of this specification (i.e. Operators that can fulfill a [service binding request](#request-service-binding)) can choose to provide a mechanism where the requester provides the mapping of existing `status` properties to their correspoding property in the [Service Binding Schema](#Service-binding-schema).  This should be a temporay state before the bindable service embraces at least the minimum requirement of this specification.
 
-*  What is the schema of a Service Binding object / structure.
-*  What metadata goes along that describes it
+#### Recommended
+In addition to the minimum set above, a bindable service **SHOULD** provide:
+* a ConfigMap that describes metadata associated with each of the items referenced in the Secret.  The bindable service should also provide a reference to this ConfigMap using one of the patterns discussed [below](#pointer-to-binding-data).
 
-### 3.  Request service binding
+The key/value pairs insides this ConfigMap are:
+* A set of `Metadata.<property>=<value>` - where `<property>` maps to one of the defined keys for this service, and `<value>` represents the description of the value.  For example, this is useful to define what format the `password` key is in, such as apiKey, basic auth password, token, etc.
 
-*  How do we request a binding from a service (assume the service has been provisioned)
 
-### 4.  Mounting binding information
+#### Pointer to binding data
 
-*  Where in the container do we mount the binding information (e.g. what is the structure of the folders / files)
-*  Consideration with clusters, namespaces, or VMs
+The reference's location and format depends on the following scenarios:
+
+1. OLM-enabled Operator: Use the `statusDescriptor` part of the CSV to mark which `status` properties reference the binding data:
+    * The reference's `x-descriptors` with one-of:
+      * ConfigMap:
+        * servicebinding:ConfigMap
+      * Secret:
+        * servicebinding:Secret
+      * Individual binding items:
+        * servicebinding:Secret:host
+        * servicebinding:Secret:port
+        * servicebinding:Secret:uri
+        * servicebinding:Secret:`<binding_property>`  (where `<binding_property>` is any property from the binding schema)
+
+2. Non-OLM Operator: - An annotation in the Operator's CRD to mark which `status` properties reference the binding data.  The value of this annotation can be specified in either [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) or [GO templates](https://golang.org/pkg/text/template/):
```

Comment:
@navidsh - I have added the text about JSONPath and Go templates to this section and `#3` as well.  I _think_ it doesn't apply to section `#1` because that `x-descriptor` is already put below the corresponding `status` key in the CRD, so it shouldn't need a path?   Let me know if that's incorrect.